src/Functions/array/FunctionArrayMapped.h

```diff
@@ -219,17 +231,65 @@ class FunctionArrayMapped : public IFunction
                                                           recursiveRemoveLowCardinality(array_type->getNestedType()),
                                                           array_with_type_and_name.name));
             }
+            if (Impl::isFolding())
```

Comment:
Judging by the amount of these `if`s, the `FunctionArrayMap` interface is not suitable for `arrayFold`. Maybe it makes sense to just copy it? It's 200 lines of mostly boilerplate.